ckanext/ckanext-apply_permissions_for_service/ckanext/apply_permissions_for_service/logic/auth.py

```diff
@@ -10,10 +10,12 @@ def service_permission_application_show(context, data_dict):
 
     permission_application_id = toolkit.get_or_bust(data_dict, 'id')
     application = model.ApplyPermission.get(permission_application_id).as_dict()
-    organization_id = application.get('organization')
+    organization = application.get('organization')
+    target_organization = application.get('target_organization')
     membership_organizations = toolkit.get_action('organization_list_for_user')(context, {'permission': 'read'})
 
-    if organization_id in [org.get('id') for org in membership_organizations]:
+    if any(True for x in [org.get('id') for org in membership_organizations]
+           if x in (organization['id'], target_organization['id'])):
```

Comment:
Potential None-error. organization and target_organization are fetched as optional with get, but not checked for None-value. Either use application['organization'] to denote the value must be in the dict or handle the None-value before using it here.